### PR TITLE
Settings Edit Form

### PR DESCRIPTION
**Pull request by lynguyxn**
Co-authored-by: Anson Lee lee.anson99@gmail.com

### Administrative Info
Monday Board ID: 1181559666
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Created frontend settings edit user form
- Modified backend edit route to solve comparing objects issue (e.g. passed in role json did not equal editedUser[field] even though it wasn't changed on the developer account which then created a permissions error with the edit route)

### Testing
How did you confirm your changes worked? 
- Updated President and Developer users on frontend, and checked it updated accordingly in database

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

President (admin) view:
image
image

Developer (non-admin) view:
image
image

### CODE REVIEW

File: frontend/src/components/UserEdit.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  nonEdit: {
+    marginBottom: theme.spacing(2),
+  },
+  textCont: {
+    textAlign: "left",
+    padding: "5%",
+  },
+  card: {
+    maxWidth: 440,
+    // marginLeft: theme.spacing(6),
```
Comment (justinyaodu) [inline frontend/src/components/UserEdit.js:42]: Can this be deleted?

File: frontend/src/pages/Settings.js [hunk 2]
```diff
@@ -29,7 +28,13 @@ const useStyles = makeStyles((theme) => ({
     width: "100%",
   },
   card: {
-    marginBottom: theme.spacing(5),
+    maxWidth: 440,
+    // marginRight: theme.spacing(6),
```
Comment (justinyaodu) [inline frontend/src/pages/Settings.js:32]: Can this be deleted?

**Review by justinyaodu — APPROVED** [review]
Good stuff! I left a few minor comments.

**Review by hahsu — APPROVED** [review]
lgtm, thanks!

**Pull request merged**